models/tokens/optimism/tokens_optimism_erc20_generated.sql

```diff
@@ -12,53 +12,46 @@ SELECT contract_address, symbol, MIN(decimals) AS decimals, token_type, token_ma
 FROM (
 
     SELECT
-    LOWER(l2_token) AS contract_address, l1_symbol AS symbol, l1_decimals as decimals
+    l2_token AS contract_address, l1_symbol AS symbol, l1_decimals as decimals
     , 'underlying' as token_type, 'l2 bridge mapping' AS token_mapping_source
     FROM {{ ref('tokens_optimism_erc20_bridged_mapping') }}
     WHERE l1_symbol IS NOT NULL
-    GROUP BY 1,2,3
 
     UNION ALL
 
     SELECT
-    LOWER(atoken_address) AS contract_address, atoken_symbol AS symbol, atoken_decimals as decimals
+    atoken_address AS contract_address, atoken_symbol AS symbol, atoken_decimals as decimals
     , 'receipt' as token_type, 'aave factory' AS token_mapping_source
     FROM {{ ref('aave_v3_tokens') }}
       WHERE blockchain = 'optimism'
-    GROUP BY 1,2,3
     
     UNION ALL
 
     SELECT
-    LOWER(atoken_address) AS contract_address, atoken_symbol AS symbol, atoken_decimals as decimals
+    atoken_address AS contract_address, atoken_symbol AS symbol, atoken_decimals as decimals
     , 'receipt' as token_type, 'the granary factory' AS token_mapping_source
     FROM {{ ref('the_granary_optimism_tokens') }}
       WHERE blockchain = 'optimism'
-    GROUP BY 1,2,3
     
     UNION ALL
 
     SELECT
-    LOWER(vault_token) AS contract_address, vault_symbol AS symbol, 18 as decimals
+    vault_token AS contract_address, vault_symbol AS symbol, 18 as decimals
     , 'receipt' as token_type, 'yearn vault factory' AS token_mapping_source
     FROM {{ ref('yearn_optimism_vaults') }}
       WHERE blockchain = 'optimism'
-    GROUP BY 1,2,3
     
   ) a
   GROUP BY contract_address, symbol, token_type, token_mapping_source --get uniques & handle if L2 token factory gets decimals wrong
 )
 
-SELECT LOWER(contract_address) AS contract_address
+SELECT contract_address
       , symbol
       , decimals
       , token_type
       , token_mapping_source
 
     FROM (
-      SELECT contract_address, symbol, decimals, token_type, token_mapping_source,
-      -- ensure we don't have dupes with different symbols
-        ROW_NUMBER() OVER (PARTITION BY contract_address ORDER BY decimals ASC NULLS LAST, symbol ASC NULLS LAST) AS token_rank
+      SELECT contract_address, symbol, decimals, token_type, token_mapping_source
```

Comment:
Why do we remove the deduplication logic?

Reply:
Agree, this should be added back in.